apidef/importer/wsdl.go

```diff
@@ -0,0 +1,430 @@
+	NS_SOAP12 = "http://schemas.xmlsoap.org/wsdl/soap12/"
+	NS_HTTP   = "http://schemas.xmlsoap.org/wsdl/http/"
+)
+
+const (
+	PROT_HTTP    = "http"
+	PROT_SOAP    = "soap"
+	PROT_SOAP_12 = "soap12"
+)
+
+type WSDL struct {
+	Services []*WSDLService `xml:"http://schemas.xmlsoap.org/wsdl/ service"`
+	Bindings []*WSDLBinding `xml:"http://schemas.xmlsoap.org/wsdl/ binding"`
+}
+
+type WSDLService struct {
+	Name  string      `xml:"name,attr"`
+	Ports []*WSDLPort `xml:"http://schemas.xmlsoap.org/wsdl/ port"`
+}
+
+type WSDLPort struct {
+	Name    string      `xml:"name,attr"`
+	Binding string      `xml:"binding,attr"`
+	Address WSDLAddress `xml:"address"`
+}
+
+type WSDLAddress struct {
+	Location string `xml:"location,attr"`
+}
+
+type WSDLBinding struct {
+	Name                string           `xml:"name,attr"`
+	Operations          []*WSDLOperation `xml:"http://schemas.xmlsoap.org/wsdl/ operation"`
+	Protocol            string
+	Method              string
+	isSupportedProtocol bool
+}
+
+type WSDLOperation struct {
+	Name             string `xml:"name,attr"`
+	Endpoint         string
+	IsUrlReplacement bool
+}
+
+func (b *WSDLBinding) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	//Get value of name attribute
+	for _, attr := range start.Attr {
+		if attr.Name.Local == "name" {
+			b.Name = attr.Value
+			break
+		}
+	}
+
+	if b.Name == "" {
+		return errors.New("Binding name is empty. Malformed wsdl")
+	}
+
+	//Fetch protocol specific data
+	//If soap/soap12 is used, set Method to POST
+	//If http is used, get value of verb attribute
+	//If any other protocol is used, then skip
+	for {
+		tok, err := d.Token()
+		if err != nil {
+			log.Error("Error will parsing wsdl file: ", err)
+			return err
+		}
+
+		switch t := tok.(type) {
+		case xml.StartElement:
+			{
+				switch t.Name.Local {
+				case "binding":
+					{
+						switch t.Name.Space {
+						case NS_SOAP, NS_SOAP12:
+							{
+								b.isSupportedProtocol = true
+								if t.Name.Space == NS_SOAP {
+									b.Protocol = PROT_SOAP
+								} else {
+									b.Protocol = PROT_SOAP_12
+								}
+
+								//Get transport protocol
+								//TODO if transport protocol is different from http
+								var transport string
+								for _, attr := range t.Attr {
+									if attr.Name.Local == "transport" {
+										transport = attr.Value
+										break
+									}
+								}
+								parts := strings.Split(transport, "/")
+								if parts[len(parts)-1] == "http" {
+									b.Method = http.MethodPost
+								} else {
+									b.isSupportedProtocol = false
+								}
+
+							}
+						case NS_HTTP:
+							{
+								b.isSupportedProtocol = true
+								b.Protocol = PROT_HTTP
+								for _, attr := range t.Attr {
+									if attr.Name.Local == "verb" {
+										b.Method = attr.Value
+										break
+									}
+								}
+
+							}
+						default:
+							{
+								log.Debug("Unsupported binding protocol is used:", t.Name.Space, ":", t.Name.Local)
```

Comment:
Perhaps use similar syntax to the other log lines, should be safe as these two `Name` fields are `string`:
```go
log.Debugf("Unsupported binding protocol is used: %s : %s", t.Name.Space, t.Name.Local)
```